chaosaws/ec2/actions.py

```diff
@@ -59,13 +59,14 @@ def stop_instances(instance_ids: List[str] = None, az: str = None,
 
     client = aws_client('ec2', configuration, secrets)
 
-    if az:
-        filters = [{'Name': 'availability-zone', 'Values': [az]}]
-        instance_ids = list_instance_ids(filters, client)
+    if not instance_ids:
+        if az:
```

Comment:
This line is actually unecessary because we already know at this stage that  ̀az` cannot be None.

Reply:
What I meant is, you cannot have instance_ids and az to be None (this tested at the beginning of the function, right?). If you don't have instance_ids then you must have an az at that stage. So the if on line 63 is not necessary I believe. You only keep if `not instance_ids`.

Reply:
Oh! Yeah right! Corrected in few minutes.
(Sorry, was thinking you were talking about L62 ><)